### PR TITLE
move model-mesh to opendatahub/manifests

### PR DESCRIPTION
**Pull request by VedantMahabaleshwarkar**
## Description

- copied over the latest model-mesh manifests from `odh-manifests/model-mesh/odh-modelmesh-controller` and put them in `opendatahub/manifests` along with nessary changes to work with the new directory structure  
- separated out modelmesh manifests from odh-model-controller manifests to enable the opendatahub-operator fetching manifests from each individual repo 

## Testing 
The testing for this PR is to be done along with the testing for [this](https://github.com/opendatahub-io/odh-model-controller/pull/92) and [this](https://github.com/opendatahub-io/opendatahub-operator/pull/580) PR since they target the same set of overall changes. 
The ODH operator image has been built with the following changes : 
-  odh-model-controller manifests are pre built from https://github.com/opendatahub-io/odh-model-controller/tree/manifests_operator_v2 which has equivalent changes to this repo, except that it deploys the image `quay.io/vedantm/odh-model-controller:remove-kserve-flag` which contains the changes for this PR. 
- model-mesh manifests are pre built from https://github.com/opendatahub-io/modelmesh-serving/tree/manifests_transition_v2 and the manfest location is opendatahub/manifests. These manifests are now no longer bundled with odh-model-controller manifests 
- opendatahub-operator reconciliation for kserve and modelmesh allow a common deployment of odh-model-controller

#### ModelMesh testing
- cluster login
- `git clone git@github.com:VedantMahabaleshwarkar/opendatahub-operator.git`
- `cd opendatahub-operator` 
- switch to branch `manifests_transition_mlserving`
- `make deploy -e IMG=quay.io/vedantm/opendatahub-operator:latest` 
- This will install the ODH operator in NS `opendatahub-operator-system` with the image `quay.io/vedantm/opendatahub-operator:latest`
- create the following DSC in NS `opendatahub` 
```
apiVersion: datasciencecluster.opendatahub.io/v1
kind: DataScienceCluster
metadata:
  name: example
spec:
  components:
    modelmeshserving:
      managementState: Managed
```
- deploy and test a model with modelmesh 

#### Kserve testing
```
Note: Cleanup kserve CRDs before testing modelmesh
```
- cluster login
- install ODH Kserve + dependencies stack by following [instructions](https://github.com/opendatahub-io/caikit-tgis-serving/blob/main/demo/kserve/install-manual.md)
```
Note: Only install the dependencies, the ODH operator will be a custom install according to next steps
```
- `git clone git@github.com:VedantMahabaleshwarkar/opendatahub-operator.git`
- `cd opendatahub-operator` 
- switch to branch `manifests_transition_mlserving`
- `make deploy -e IMG=quay.io/vedantm/opendatahub-operator:latest` 
- This will install the ODH operator in NS `opendatahub-operator-system` with the image `quay.io/vedantm/opendatahub-operator:latest`
- create the following DSC in NS `opendatahub` 
```
apiVersion: datasciencecluster.opendatahub.io/v1
kind: DataScienceCluster
metadata:
  name: default
spec:
  components:
    kserve:
      managementState: Managed
```
- deploy and test a kserve model using [instructions](https://github.com/opendatahub-io/caikit-tgis-serving/blob/main/demo/kserve/deploy-remove-scripts.md)

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: opendatahub/manifests/base/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../default
+  - ../prometheus
+  - ../overlays/odh
+
+commonLabels:
+  app.kubernetes.io/part-of: model-mesh
```
Comment (israel-hdez) [inline opendatahub/manifests/base/kustomization.yaml:10]: Is this right to be here?
I am not able to find it on original manifests.

File: opendatahub/manifests/base/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      apiVersion: v1
+      kind: ConfigMap
+      name: mesh-parameters
+  - fieldref:
+      fieldPath: data.monitoring-namespace
+    name: monitoring-namespace
+    objref:
+      apiVersion: v1
+      kind: ConfigMap
+      name: mesh-parameters
```
[the remark this answers is not in the document]
Comment (VedantMahabaleshwarkar) [reply opendatahub/manifests/base/kustomization.yaml:33]: yeah we should remove this from the file, however that should be a separate PR imo. I did not want to have both changes in 1 PR

File: opendatahub/manifests/base/kustomization.yaml (hunk 1 above)
Comment (VedantMahabaleshwarkar) [reply opendatahub/manifests/base/kustomization.yaml]: looks like it was recently removed in [this commit](https://github.com/opendatahub-io/modelmesh-serving/commit/bdfdca83fce58703c42392e0cb8d7daecd5aa519). Removed it from this PR too.

File: opendatahub/manifests/base/kustomization.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (israel-hdez) [reply opendatahub/manifests/base/kustomization.yaml:31]: I was meaning, that it doesn't seem to be used in this PR.
Or is it? If so, where is being used?

[the remark this answers is not in the document]
Comment (VedantMahabaleshwarkar) [reply opendatahub/manifests/base/kustomization.yaml]: removed

**Comment by vaibhavjainwiz** [comment]
@VedantMahabaleshwarkar Rather them maintaining manifest in `opendatahub/manifests` folder could we simply maintain them inside `/config` folder itself so that it would be more allign with the way we are maintaing manifest for **odh-model-controller** and **kserve**.

**Comment by VedantMahabaleshwarkar** [comment]
replaced by https://github.com/opendatahub-io/modelmesh-serving/pull/237

**Pull request closed without merging**